operations.go

```diff
@@ -147,6 +148,52 @@ func (p *PingOp) Operate(t testing.TB) {
 	}
 }
 
+// GetSystemTime returns system time
+func (o *Operations) GetSystemTime() (uint64, error) {
+	return operations.GetSystemTime(context.Background(), o.dut)
+}
+
+// NewFactoryReset creates a new Factory Reset.
+func (o *Operations) NewFactoryReset() *FactoryReset {
+	return &FactoryReset{
+		dut:       o.dut,
+		factoryOs: false,
+		zeroFill:  false,
+	}
+}
+
+// FactoryReset enables resetting the device to factory settings
+type FactoryReset struct {
+	dut       binding.DUT
+	factoryOs bool
+	zeroFill  bool
+}
+
+// WithFactoryOS instructs the Target if it needs to rollback the OS to the same version as it shipped
+func (s *FactoryReset) WithFactoryOS(factoryOs bool) *FactoryReset {
+	s.factoryOs = factoryOs
+	return s
+}
+
+// WithZeroFill instructs the Target  if it has to zero fill persistent storage state data.
+func (s *FactoryReset) WithZeroFill(zeroFill bool) *FactoryReset {
+	s.zeroFill = zeroFill
+	return s
+}
+
+// String representation of the method
+func (s *FactoryReset) String() string {
+	return fmt.Sprintf("FactoryResetOp%+v", *s)
+}
+
+// Operate performs the FactoryReset operation.
+func (s *FactoryReset) Operate(t testing.TB) error {
+	t.Helper()
+	err := operations.FactoryReset(context.Background(), s.dut, s.factoryOs, s.zeroFill)
+	return err
```

Comment:
Shouldn't return errors from methods that accept testing.T. Follow the example of the other Operate functions and translate the error into a t.Fatal failure

Reply:
I have added t.Fatal